### ISSUE TITLE
Notice when the script moveMouseToNavigatorObject in globalCommands can't move the mouse to the object.

### ISSUE THREAD

**Issue by javidominguez** [nvaccess/nvda#7401]
Sometimes the script_moveMouseToNavigatorObject does not work because the object in the browser is inaccessible to the mouse cursor. Maybe because its location is outside of the visible screen or because it is covered by another object. The script should warn it to avoid clicking in a undesired place.

I think it is easy to implement. The code could be something like this:

>		if api.getDesktopObject().objectFromPoint(x,y) != obj:
>			ui.message(_("Can't move the cursor here"))
>			return

[script_moveMouseToNavigatorObject.py.zip](https://github.com/nvaccess/nvda/files/1150598/script_moveMouseToNavigatorObject.py.zip)

**Comment by derekriemer** [comment]
Does anyone at NV Access have an issue with this being put in a PR?

**Comment by jcsteh** [comment]
I agree that users potentially clicking the wrong things is nasty. Unfortunately, it's not always this simple:

1. Moving the mouse to an object might take you to a descendant of that object. That's perfectly valid for a container. It can even occur on something like a list item if, for example, it was represented by a graphic and some text (or multiple pieces of text). Walking the ancestors might help catch this.
2. Moving the mouse to the object might cause another object to appear in its place/on top of it; e.g. tool tip, drop down, overlay. We still want users to have the ability to access such things.
3. The mouse might actually move to the right place, but the application might not support hit testing correctly. In that case, retrieving the object at the coordinates will fail or return the wrong object.
4. There are cases when it's useful to know what's obscuring an object (when one object is behind another). Admittedly, this is a corner case and is probably only really relevant to developers, so we could cover this case by using a config option (not exposed in GUI).

@feerrenrut, how likely is point 2? Do drop downs, tool tips, etc. ever obscure the centre of an object or do they always appear near it without obscuring it?

At the very least, to even consider this, we'd need the ancestor check noted in point 1 above. We could also fail early for coordinates outside the object's HWND. I'm still concerned about the remaining points, though.

@MichaelDCurran, thoughts?

**Comment by jcsteh** [comment]
P3 because it'd be nice if users weren't able to click the wrong things unintentionally. However, this still needs a fair chunk of investigation.

**Comment by feerrenrut** [comment]
> how likely is point 2? Do drop downs, tool tips, etc. ever obscure the centre of an object or do they always appear near it without obscuring it?

There probably are some rare cases where moving the mouse over the object might change what is there, in general I think this would be considered bad UX though. The object is unlikely to be obscured (especially by a tooltip, which would usually appear near, but offset from the mouse).

**Comment by Adriani90** [comment]
for an user this would just add more verbosity which in my view is not necessary. If you know a website or an application, you also know the labels of the objects. If the mouse is moved to an object next to a desired one, NVDA would reports its label and the user would know that it is near to the desired object. After that, just moving the mouse very slowly around increases the chance to land on the right object. I am dealing with many applications which have this issue at my workplace. It works very well if you know the applications. For beginer users this kind of navigation is far from usual and most of them would never try to navigate like this in an applicatoin. The rpobability for an experienced user to click on a wrong object is quite low in my view so we do not really need the extra verbosity.

**Comment by Adriani90** [comment]
moving mouse to the object is quite unreliably in a lot of places. Having an alert every time would decrease efficiency after a while.

**Comment by Adriani90** [comment]
@CyrilleB79, @XLTechie could you please also review the arguments above? Do you think this feature request can be closed?

**Comment by XLTechie** [comment]
I wish there was a reliable test case or two for this.
I would also like to know how Jaws and Narrator handle this situation.

I have had this problem more than a few times. It is usually possible to know it 
might be what's going on, but such clicks, in my experience, haven't been very 
bad in their side effects.
It is a legitimate concern though.

However, also valid concerns, are everything that @jcsteh pointed out.

Just thinking about it briefly, I can't think of any user friendly way to do 
this that wouldn't be just as unreliable as the current situation. And worse, as 
@Adriani90 mentioned in an earlier comment: anything we do with this may 
be more verbose, and maybe wrongly so at that.

Without test cases, it is very hard to determine whether trying any of what 
has been suggested, is even workable.

I would think it might be a problem worth solving, if any of the solutions 
weren't worse than the problem itself.

Vague ideas:

Maybe play a sound if the object (or a child) was reached, and a different one 
if it wasn't? (I am thinking ascending/descending tones here.)

Maybe a note in the user guide, that indicates the possibility of unreliable 
mouse pointer routing?

Those are my only two ideas short of closing this as a can't fix.

**Comment by CyrilleB79** [comment]
I do not know if the solution described in this feature request is the good one.

However the problem it tries to solve is a real and frequent one for me. Note: I am a Magnifier user; I do not know if it makes thing worse.

To work on this topic, I think that we should:
- first Identify as precisely as possible the situations where mouse routing does not work as expected, e.g. providing STRs
- then understand the root cause of it
- finally think to a solution

Are there other tickets on the same topic?
